packages/botonic-cli/src/commands/deploy.ts

```diff
@@ -54,28 +59,49 @@ Uploading...
     }),
   }
 
-  static args = [{ name: 'bot_name' }]
+  static args = [{ name: 'provider', options: Object.values(CLOUD_PROVIDERS) }]
 
   private botonicApiService: BotonicAPIService = new BotonicAPIService()
   private botName: string | undefined = undefined
   private telemetry = new Telemetry()
 
   /* istanbul ignore next */
   async run(): Promise<void> {
-    const { flags } = this.parse(Run)
-    this.telemetry.trackDeploy()
-    npmCommand = flags.command
-    this.botName = flags.botName
-    const email = flags.email
-    const password = flags.password
-    if (email && password) await this.login(email, password)
-    else if (!this.botonicApiService.oauth) await this.signupFlow()
-    else if (this.botName) {
-      await this.deployBotFromFlag(this.botName)
-    } else await this.deployBotFlow()
+    const { flags, args } = this.parse(Run)
+    const provider = args.provider
+    if (provider && provider !== CLOUD_PROVIDERS.HUBTYPE) {
+      this.deployToProvider(provider)
+    } else {
```

Comment:
This logic is divided inconsistently between `run()` and `deployToProvider()` functions. Better do something like:

```
async run(): Promise<void> {
  const { flags, args } = this.parse(Run)
  const provider = args.provider || CLOUD_PROVIDERS.HUBTYPE
  // TODO -> In a next iteration it would be cool to get a prompt asking which provider we prefer (if it's the 1st deploy) or getting the provider from `.botonic.json` (after 1st deploy)
  this.telemetry.trackDeploy1_0({ provider })
  console.log(`Deploying to ${provider}...`)
  console.log('This can take a while, do not cancel this process.')
  if (provider === CLOUD_PROVIDERS.AWS) {
    this.deployAWS(...)
  } else if (provider === CLOUD_PROVIDERS.HUBTYPE) {
    this.deployHubtype(...)
  }
}

async deployAWS(): Promise<void> { }
async deployHubtype(): Promise<void> { /* Current code */ }

```

Reply:
I agree, separated the deploy paths consistently 👍